### PR TITLE
Dev/crjr/7632 removeitem combobox

### PR DESCRIPTION
**Pull request by CesarRabelo**
GitHub Issue (If applicable): closes #
#7632 

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [X] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [X] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [X] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs [hunk 1]
```diff
@@ -934,13 +969,17 @@ void LocalCleanupContainer(object container)
 
 					LocalCleanupContainer(container);
 					RequestLayoutPartial();
+					SaveContainersForIndexRepair(args.OldStartingIndex, -args.OldItems.Count);
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs:972]: The way this method is used in ListViewBase is to call it _before_ actually applying the collection change, to save the prior state. Here that would mean calling it before `ItemsPanel.Children.RemoveAt`.

**Review by davidjohnoliver — COMMENTED** [review]
The changes seem to cause other selection-related tests to fail:

image


https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=39982&view=ms.vss-test-web.build-test-results-tab

File: src/Uno.UI/UI/Xaml/Controls/Primitives/Selector.cs [hunk 2]
```diff
@@ -164,7 +164,8 @@ internal virtual void OnSelectedItemChanged(object oldSelectedItem, object selec
 			if (updateItemSelectedState)
 			{
 				TryUpdateSelectorItemIsSelected(oldSelectedItem, false);
-				TryUpdateSelectorItemIsSelected(selectedItem, true);
+				if (SelectedValue != selectedItem)
+					TryUpdateSelectorItemIsSelected(selectedItem, true);
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/Primitives/Selector.cs:168]:
```suggestion
				if (SelectedValue != selectedItem)
				{
					TryUpdateSelectorItemIsSelected(selectedItem, true);
				}
```

File: src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs (hunk 1 above)
Comment (CesarRabelo) [reply src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs]: Resolved according your suggestion.

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/ComboBox/ComboBox_RemoveSelectedIndex.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Navigation;
+
+// The User Control item template is documented at https://go.microsoft.com/fwlink/?LinkId=234236
+
+namespace UITests.Shared.Windows_UI_Xaml_Controls.ComboBox
+{
+	[SampleControlInfo("ComboBox", "ComboBox_RemoveSelectedIndex")]
```
Comment (carldebilly) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/ComboBox/ComboBox_RemoveSelectedIndex.xaml.cs:21]:
```suggestion
	[Sample]
```

File: src/Uno.UI.Tests/Windows_UI_XAML_Controls/ComboBoxTests/Given_ComboBox.cs [hunk 4]
```diff
@@ -121,5 +121,25 @@ public void When_Removed_Item_Is_The_SelectedIndex()
...
+		{
+			var comboBox = new ComboBox();
+			string[] items = new string[] { "Item #0", "Item #1", "Item #2", "Item #3", "Item #4", "Item #5" };
+			foreach (string item in items)
+			{
+				ComboBoxItem ni = new ComboBoxItem { Content = item, IsSelected = item == "Item #2" };
+				comboBox.Items.Add(ni);
+			}
+
+			Assert.AreEqual<int>(2, comboBox.SelectedIndex);
```
Comment (carldebilly) [inline src/Uno.UI.Tests/Windows_UI_XAML_Controls/ComboBoxTests/Given_ComboBox.cs:136]: The `<int>` generic there is not required and can be inferred by the compiler.

File: src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs [hunk 5]
```diff
@@ -62,6 +62,41 @@ public partial class ItemsControl : Control, IItemsControl
...
+					_containersForIndexRepair.Add(container, currentIndex + indexChange);
+				}
+			}
+		}
+
+		/// <summary>
+		/// Sets the indices of stored materialized containers to the appropriate index after
+		/// collection change.
+		/// </summary>
+		private void RepairIndices()
```
[the remark this answers is not in the document]
Comment (CesarRabelo) [reply src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs]: Merger executed as suggested.

**Pull request closed without merging**